src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -353,15 +353,18 @@ export class HttpConnection implements IConnection {
         const transports = negotiateResponse.availableTransports || [];
         for (const endpoint of transports) {
             try {
-                const transport = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
-                if (typeof transport === "number") {
-                    this.transport = this.constructTransport(transport);
+                const transportOrError = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
+                if (typeof transportOrError === "number") {
+                    this.transport = this.constructTransport(transportOrError);
                     if (!negotiateResponse.connectionId) {
```

Comment:
I see, so we always want to renegotiate for transport!.connect failures, because we don't want to risk reusing the same connectionid. But in the case that the client doesn't support the transport, there's no reason not to use the connectionid we already have, because we have tried using it yet.

My question is if getNegotiationResponse itself fails, not the transport, will the client move on to the next transport in the list and try renegotiating? That seems super broken. If any negotiate request fails, start should probably just fail. I'm fine leaving this PR as is and filing a separate issue though.

Reply:
Good observation, we should probably jump out of the loop in that case